fs/layer/layer.go

```diff
@@ -345,9 +345,12 @@ func (r *Resolver) Resolve(ctx context.Context, hosts []docker.RegistryHost, ref
 	if err != nil {
 		return nil, fmt.Errorf("failed to read layer: %w", err)
 	}
-	disableXattrs := getDisableXAttrAnnotation(sociDesc)
+	disableXAttrs := getDisableXAttrAnnotation(sociDesc)
+	if r.config.ForceGetXAttrs {
+		disableXAttrs = false
+	}
```

Comment:
Slight optimization we can use short circuiting to avoid parsing the string value set in layer annotations when the configuration is to force get xattrs.

```suggestion
        disableXAttrs := r.config.ForceGetXAttrs && getDisableXAttrAnnotation(sociDesc)
```

Reply:
Good idea, though the first variable should be inverted, since the only condition where it is true is if we do not want to force GetXAttrs and the annotation is also true.

This probably goes to show how confusing this bit is, though. I added an extra comment to hopefully clarify for future readers.